### ISSUE TITLE
Searching for bflix is dead.

### ISSUE THREAD

**Issue by lowpolyv2** [JDALab/film-central#14]
Querying no longer works on the bflix website for some reason, so searching for the scraper is currently broken.

**Comment by lowpolyv2** [comment]
@Fraei2 found a way around this 🙂

![Screenshot_20240928_221455](https://github.com/user-attachments/assets/9f543acd-5c94-4a9b-9379-a54dd53a8356)

content type **must** be **exactly** as set then you make a POST request to that URL with a text body like so and also the search query will need to be URL encoded.

I'm too busy these days so anyone feel free to implement this, I would say just keep the old search logic in case they bring searching back via the current solution.

**Comment by yawnbo** [comment]
Is nitestv working at all still? I'd love to help but can't get a server response through any means. However, one of the mirrors, bflix.sh, is working and I can try working on that

**Comment by lowpolyv2** [comment]
blfix.sh doesn't work too right now

**Comment by lowpolyv2** [comment]
It's probably down for the time being, given it some time.

**Comment by yawnbo** [comment]
weird, bflix.sh is still up for me, nites.nz doesn't respond, and nites.co's domain is up for sale on a redirect LMAO

**Comment by lowpolyv2** [comment]
> weird, bflix.sh is still up for me, nites.nz doesn't respond, and nites.co's domain is up for sale on a redirect LMAO

yeah LMAO I also saw nites.co on sale too but uhhh... the bflix.sh domain also seems to be out of ownership...

![image](https://github.com/user-attachments/assets/4a5e0498-53b7-47e9-9a10-c117f6c1ead6)

**Comment by yawnbo** [comment]
That's really weird, I'm able to access and search on it still, 
![image](https://github.com/user-attachments/assets/31521410-15fe-4f99-8311-15b05fa3e222)

**Comment by lowpolyv2** [comment]
Weird probably DNS. Btw in case you didn't already know bflix was rebranded into nites a long time ago and bflix.sh is not their official domains, so even if we switched the domain to bflix.sh it wouldn't work as this seems like an entirely different website that will need a different scraper to scrape.

**Comment by yawnbo** [comment]
yeah I was looking around yesterday and it's def a mirror. I'll look around for how bad setting up a new scraper would be and maybe work on it

**Comment by yawnbo** [comment]
but it's def something with DNS because cloudflare returns a 523 error or something

**Comment by lowpolyv2** [comment]
@Fraei2 https://catflix.su/ seems to be a good site to scrape, if you're free I would say have a look at it. 
So far it seems like all you need is the referer set and streams play on mpv no problem, now getting to the stream URL you'll have to figure out.

Peace ✌️

**Comment by yawnbo** [comment]
oops forgot i said i would do bflix until this, i did a lookaround on the upstream and found that mp2t fragments are at "https://s.kekw.site/xgdkvyp/uwu_XXX.EXT" where XXX is 007-999, EXT is a random web file extension (css, js, html) and "xgdkvyp" is the id (in this case i was testing with interstellar) I have no clue how to figure out the id and the rest though. [here](https://pastebin.com/7CB9J5uT) is one of the ajax calls in the js file that i assume would help

**Comment by duck-not-found** [comment]
hey i made a script that scrapes the embed links and the refferals but i need help

**Comment by lowpolyv2** [comment]
> hey i made a script that scrapes the embed links and the refferals but i need help
> 

How does this script function? Does it use beautiful soup?

**Comment by duck-not-found** [comment]
not exactly, it scrapes the catflix website and finds a turbovid.eu/embed link (sorry for late response i was distracted trying to make this work) but it gives the embed link and works on browser

**Comment by duck-not-found** [comment]
it only uses import requests and import re

**Comment by lowpolyv2** [comment]
> it only uses import requests and import re 
> 

Oh that's perfect. Yeah feel free to send your script here if you want.

**Comment by duck-not-found** [comment]
import requests
import re

def get_movies(search_term):
    url = f'https://catflix.su/?s={search_term}'
    response = requests.get(url)
    pattern = r'https://catflix\.su/movies/(?!movies/)[^"]+'
    matches = re.findall(pattern, response.text)
    return matches

def get_turbovid_url(url):
    response = requests.get(url)
    pattern = r'https://turbovid\.eu/embed/[^"]+'
    matches = re.findall(pattern, response.text)
    return matches

search_term = input("Enter your search term: ")
matches = get_movies(search_term)

if matches:
    print("Found the following movie URLs:")
    for i, match in enumerate(matches, start=1):
        title = match.replace("https://catflix.su/movies/", "").replace("/", "").replace("-", " ")
        title = title.title()
        print(f"{i}. {title}")
    choice = input("Enter the number of the movie you want to watch: ")
    try:
        choice = int(choice)
        if 1 <= choice <= len(matches):
            url = matches[choice - 1]
            turbovid_urls = get_turbovid_url(url)
            if turbovid_urls:
                print("Found the following TurboVid URLs:")
                for turbovid_url in turbovid_urls:
                    print(turbovid_url)
            else:
                print("No TurboVid URLs found")
        else:
            print("Invalid choice")
    except ValueError:
        print("Invalid choice")
else:
    print("No movie URLs found")

**Comment by duck-not-found** [comment]
> > it only uses import requests and import re
> 
> Oh that's perfect. Yeah feel free to send your script here if you want.

https://github.com/duck-not-found/catflix-scraper.git
i made a repo with the code actually indented

**Comment by duck-not-found** [comment]
how's progress?

**Comment by lowpolyv2** [comment]
> how's progress?

I think the repo isn't public, I get a 404 not found.

**Comment by duck-not-found** [comment]
how about now?

**Comment by duck-not-found** [comment]
i made it public, fixed.

**Comment by lowpolyv2** [comment]
Yeah I can see it.

Okay so that's what you meant by embed URL. Sadly this alone won't really work when we get to streaming it via video players like MPV as for that we'll need to get the direct video stream URL and I think I recall looking at that embedding site before and from the way I remember they did streaming, it's gonna take a lot to reverse engineer it.

**Comment by duck-not-found** [comment]
i tried reverse engineering the embeds and all i found was some APIs and then the request responds with some long string of chars.

**Comment by lowpolyv2** [comment]
> i tried reverse engineering the embeds and all i found was some APIs and then the request responds with some long string of chars.

yep, I've been through a lot of this lmao

**Comment by lowpolyv2** [comment]
> i found the .m3u8s but they always 403 me

try making a request to that file with a referrer http header perhaps (e.g. "https://turbovid.eu"). Sometimes (well actually most of the time lmao) they deny you from streaming the file directly by checking if you have a specific referrer URL set in the referrer http header.

**Comment by duck-not-found** [comment]
huh... i'll try

**Comment by lowpolyv2** [comment]
Also I recommend a tool called httptoolkit if you wanna look deeper into these embed sites without them freaking out when you open up browser developer tools.

**Comment by duck-not-found** [comment]
curl 'https://s.kekw.site/xakv/uwu.m3u8' -H 'Referer: https://turbovid.eu/' does indeed work and does print some similarity to an m3u8 file like this one. (also that's the sonic the hedgehog 2 movie link lol) also i had to rename it to .TXT
[uwu.txt](https://github.com/user-attachments/files/18131013/uwu.txt)

**Comment by duck-not-found** [comment]
if i could somehow be able to use the links in vlc but with the header info i **_might_** be able to stream it.

**Comment by duck-not-found** [comment]
OH MY GOD I DID IT
i used ffmpeg (somehow) and made IT stream the contents and copy them into an MP4, the result (after not actually cloning the entire movie) THIS (command was:ffmpeg -headers "Referer: https://turbovid.eu/" -i https://s.kekw.site/xakv/uwu.m3u8 -c copy output.mp4 btw)

https://github.com/user-attachments/assets/12dc18ab-0ed2-4a84-93ed-bda85864f3f4

**Comment by duck-not-found** [comment]
ffmpeg -headers "Referer: https://turbovid.eu/" -i https://s.kekw.site/xakv/uwu.m3u8 -f mpegts - | mpv - voila, FIXED.

**Comment by lowpolyv2** [comment]
> if i could somehow be able to use the links in vlc but with the header info i **_might_** be able to stream it.

yeah you also can with mpv with:
```
mpv https://s.kekw.site/xakv/uwu.m3u8 --http-header-fields="Referer: https://turbovid.eu/"
```

**Comment by MisaelVM** [comment]
Hi there, guys. I think I got it working, should I open a PR?

**Comment by lowpolyv2** [comment]
> Hi there, guys. I think I got it working, should I open a PR?

Go ahead
